src/shared_modules/content_manager/src/components/CtiDownloader.hpp

```diff
@@ -85,22 +85,50 @@ class CtiDownloader : public AbstractHandler<std::shared_ptr<UpdaterContext>>
      */
     CtiBaseParameters getCtiBaseParameters(const std::string& ctiURL)
     {
-        CtiBaseParameters parameters;
+        nlohmann::json rawMetadata;
 
         // Routine that stores the necessary parameters.
-        const auto onSuccess {[&parameters](const std::string& response)
+        const auto onSuccess {[&rawMetadata](const std::string& response)
                               {
-                                  const auto responseData = nlohmann::json::parse(response).at("data");
+                                  logDebug2(WM_CONTENTUPDATER, "CTI raw metadata: '%s'", response.c_str());
+
+                                  if (!nlohmann::json::accept(response))
+                                  {
+                                      throw std::runtime_error {"Invalid CTI metadata format"};
+                                  }
 
-                                  parameters.lastOffset = responseData.at("last_offset").get<int>();
-                                  parameters.lastSnapshotLink =
-                                      responseData.at("last_snapshot_link").get<std::string>();
-                                  parameters.lastSnapshotOffset = responseData.at("last_snapshot_offset").get<int>();
+                                  rawMetadata = nlohmann::json::parse(response).at("data");
                               }};
 
         // Make a get request to the API to get the consumer offset.
         performQueryWithRetry(ctiURL, onSuccess);
 
+        // Return if interrupted.
+        if (m_spUpdaterContext->spUpdaterBaseContext->spStopCondition->check())
+        {
+            return CtiBaseParameters();
+        }
+
+        // Validate metadata.
+        for (const std::string& key : {"last_offset", "last_snapshot_link", "last_snapshot_offset"})
+        {
+            if (!rawMetadata.contains(key))
+            {
+                throw std::runtime_error {"Missing CTI metadata key: " + key};
+            }
+
+            const auto& data {rawMetadata.at(key)};
+            if (data.is_null() || (data.is_string() && data.empty()))
```

Comment:
This should be something like:
```suggestion
            if (data.is_null() || (data.is_string() && data.get_ref<const std::string&>().empty()))
```
`data.empty()`  is always `false` when `data` is a string, even if the string is empty. See [here](https://json.nlohmann.me/api/basic_json/empty/)

Reply:
Good catch! I forgot to add a UT for this scenario...
Fixed and UT added.